app/javascript/pages/country/header/header.js

```diff
@@ -77,49 +80,91 @@ const mapDispatchToProps = dispatch =>
     if (location.region) {
-      getTotalArea({ ...location });
+      getTotalExtent({ ...location, ...settings });
     }
     if (location.subRegion) {
-      getTotalArea({ ...location });
+      getTotalExtent({ ...location, ...settings });
     }
   }
 
   componentWillReceiveProps(nextProps) {
-    const { country, region, subRegion } = nextProps.location;
-    const { getTotalArea, getTreeCoverExtent } = this.props;
-    const hasCountryChanged = country !== this.props.location.country;
-    const hasRegionChanged = region !== this.props.location.region;
-    const hasSubRegionChanged = subRegion !== this.props.location.subRegion;
+    const { location, settings } = nextProps;
+    const { getTotalExtent, getTotalLoss, getPlantationsLoss } = this.props;
 
-    if (hasCountryChanged) {
-      getTotalArea({ ...nextProps.location });
-      getTreeCoverExtent({ ...nextProps.location });
+    if (!isEqual(location, this.props.location)) {
+      getTotalExtent({ ...nextProps.location, ...settings });
+      getTotalLoss({ ...nextProps.location, ...settings });
+      getPlantationsLoss({
+        ...nextProps.location,
+        ...settings,
+        indicator: 'plantations'
+      });
     }
-    if (region && hasRegionChanged) {
-      getTotalArea({ ...nextProps.location });
-      getTreeCoverExtent({ ...nextProps.location });
+  }
+
+  getHeaderDescription = () => {
+    const { locationNames, data } = this.props;
+    const extent = format('.2s')(data.extent);
+    const percentageCover = format('.1f')(data.extent / data.totalArea * 100);
+    const lossWithOutPlantations = format('.2s')(
+      data.totalLoss.area - (data.plantationsLoss.area || 0)
+    );
+    const emissionsWithoutPlantations = format('.2s')(
+      data.totalLoss.emissions - (data.plantationsLoss.emissions || 0)
+    );
+    const location = locationNames.current && locationNames.current.label;
+    let firstSentence = '';
+    let secondSentence = '';
+    if (extent) {
+      firstSentence = `
+        In 2010, <b>${location}</b> had <b>${extent}ha</b> of tree cover, extending over <b>${percentageCover}%</b> of its land area.
+      `;
+    } else {
+      firstSentence = `
+        In 2010, <b>${location}</b> had <b>${extent}ha</b> of tree cover.
+      `;
     }
-    if (subRegion && hasSubRegionChanged) {
-      getTotalArea({ ...nextProps.location });
-      getTreeCoverExtent({ ...nextProps.location });
+    if (data.totalLoss.area) {
+      secondSentence = `
+        In ${
+          data.totalLoss.year
+        }, it lost <b>${lossWithOutPlantations}ha</b> of forest ${
+          data.plantationsLoss.area ? 'excluding tree plantations' : ''
+        }, equivalent to <b>${emissionsWithoutPlantations}tonnes</b> of CO₂ of emissions.
+      `;
     }
-  }
+    const totalSentence = `${firstSentence} ${secondSentence}`;
+
+    return <p dangerouslySetInnerHTML={{ __html: totalSentence }} />;
```

Comment:
What do you think about set the `dangerouslySetInnerHTML` on the component and this method only returns `{ __html: totalSentence }`. With this approach you don't need import `React` and we leave the responsibility of painting to the component.

Reply:
Yes I think you are correct.